content/127labs/drone-k8s-helm/index.md

```diff
@@ -0,0 +1,54 @@
+---
+date: 2017-06-10T00:00:00+00:00
+title: Kubernetes Helm
+author: 127labs
+tags: [ kubernetes, helm, docker ]
+repo: 127labs/drone-k8s-helm
+logo: kubernetes_helm.svg
+image: 127labs/drone-k8s-helm
+---
+
+This Kubernetes plugin can be included into your pipelines to allow deployment with Helm.
+
+Example configuration:
+
+```yaml
+pipeline:
+  ship:
+    image: 127labs/drone-k8s-helm
+    chart: stable/dokuwiki
+    master: https://127.0.0.1:8001
+    token: super-long-token
+    release: wiki
+    skip_tls: true
+    clean_before_release: true
+    values:
+      dokuwikiEmail: $${DOKUWIKI_EMAIL}
+      dokuwikiPassword: $${DOKUWIKI_PASSWORD}
+    secrets: [dokuwiki_email, dokuwiki_password]
+```
+
+# Secret Reference
+
+Values are capable of accessing secrets by interpolating them with the `$${SECRET}` syntax. The plugin will
```

Comment:
also I noticed this in the github readme:

```
    secrets:
      - dokuwiki_email
      - dokuwiki_password
      - plugin_master
      - plugin_token
```

`PLUGIN_` is a convention used to namespace variables that are defined in the yaml. In the future we may not let users set variables with `PLUGIN_` so you may not be able to rely on using this value for user-defined secrets.

If you look at the docker plugin (for example) you will notice we source the environment variable name from `PLUGIN_` and from a more user-friendly environment variable which the end-user can use in conjunction with secrets.

```
		cli.StringFlag{
			Name:   "docker.username",
			Usage:  "docker username",
			EnvVar: "PLUGIN_USERNAME,DOCKER_USERNAME",
		},
		cli.StringFlag{
			Name:   "docker.password",
			Usage:  "docker password",
			EnvVar: "PLUGIN_PASSWORD,DOCKER_PASSWORD",
		},
```

PS sorry if this is all a bit confusing. I haven't had time to put together a best practices document for plugin authors